front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -29,30 +36,147 @@ import {
   isManaged,
   isWebsite,
 } from "@app/lib/data_sources";
+import { useVaults } from "@app/lib/swr/vaults";
+import { classNames } from "@app/lib/utils";
+import { getVaultIcon, getVaultName } from "@app/lib/vaults";
 
 const MIN_TOTAL_DATA_SOURCES_TO_GROUP = 12;
 const MIN_DATA_SOURCES_PER_KIND_TO_GROUP = 3;
+const ONLY_ONE_VAULT_PER_SELECTION = true;
 
 interface DataSourceViewsSelectorProps {
   owner: LightWorkspaceType;
   dataSourceViews: DataSourceViewType[];
+  allowedVaults?: VaultType[];
   selectionConfigurations: DataSourceViewSelectionConfigurations;
   setSelectionConfigurations: Dispatch<
     SetStateAction<DataSourceViewSelectionConfigurations>
   >;
   viewType: ContentNodesViewType;
 }
 
+function VaultSelector({
+  owner,
+  dataSourceViews,
+  allowedVaults,
+  selectionConfigurations,
+  setSelectionConfigurations,
+  viewType,
+}: DataSourceViewsSelectorProps) {
+  const dataSourceViewsByVaultId = useMemo(
+    () => _.groupBy(dataSourceViews, (dsv) => dsv.vaultId),
+    [dataSourceViews]
+  );
+
+  const { vaults, isVaultsError, isVaultsLoading } = useVaults({
+    workspaceId: owner.sId,
+  });
+
+  const [selectedVault, setSelectedVault] = useState<string>(
```

Comment:
I'd assume that the default selected vault should be the global one.

Reply:
If I understand correctly, here we pick the vault used by the first action, is that right? Why not defaulting to the global one?